ftml/src/parsing/rule/impls/block/blocks/char.rs

```diff
@@ -0,0 +1,236 @@
+        let mut mapping = HashMap::new();
+
+        for entity in &ENTITIES {
+            let key = strip_entity(entity.entity);
+            let value = entity.characters;
+
+            mapping.insert(key, value);
+        }
+
+        mapping
+    };
+}
+
+pub const BLOCK_CHAR: BlockRule = BlockRule {
+    name: "block-char",
+    accepts_names: &["char", "character"],
+    accepts_special: false,
+    accepts_newlines: false,
+    parse_fn,
+};
+
+fn parse_fn<'r, 't>(
+    log: &slog::Logger,
+    parser: &mut Parser<'r, 't>,
+    name: &'t str,
+    special: bool,
+    in_head: bool,
+) -> ParseResult<'r, 't, Elements<'t>> {
+    debug!(log, "Parsing character / HTML entity block"; "in-head" => in_head);
+
+    assert_eq!(special, false, "Char doesn't allow special variant");
+    assert_block_name(&BLOCK_CHAR, name);
+
+    // Parse the entity and get the string
+    let string = parser.get_head_value(&BLOCK_CHAR, in_head, parse_entity)?;
+
+    ok!(Element::Text(string))
+}
+
+fn parse_entity<'r, 't>(
+    parser: &Parser<'r, 't>,
+    argument: Option<&'t str>,
+) -> Result<Cow<'t, str>, ParseWarning> {
+    let argument = match argument {
+        Some(arg) => strip_entity(arg),
+        None => return Err(parser.make_warn(ParseWarningKind::BlockMissingArguments)),
+    };
+
+    match find_entity(argument) {
+        Some(string) => Ok(string),
+        None => Err(parser.make_warn(ParseWarningKind::BlockMalformedArguments)),
+    }
+}
+
+/// Find the string corresponding to the passed entity, if any.
+fn find_entity(entity: &str) -> Option<Cow<str>> {
+    // Named entity
+    if let Some(result) = ENTITY_MAPPING.get(entity) {
+        return Some(cow!(result));
+    }
+
+    // Hexadecimal entity
+    if let Some(value) = entity.strip_prefix("#x") {
+        if let Some(result) = get_char(value, 16) {
+            return Some(result);
+        }
+    }
+
+    // Decimal entity
+    if let Some(value) = entity.strip_prefix('#') {
+        if let Some(result) = get_char(value, 10) {
+            return Some(result);
+        }
+    }
+
+    // Not found
+    None
+}
+
+/// Gets the appropriate character from the number specified in the string.
+///
+/// Using the passed radix, it gets the integer value, then finds the appropriate
+/// character, if one exists.
+///
+/// Then converts the character into a string with only that value.
+fn get_char(value: &str, radix: u32) -> Option<Cow<str>> {
+    let codepoint = match u32::from_str_radix(value, radix) {
+        Ok(codepoint) => codepoint,
+        Err(_) => return None,
+    };
+
+    let ch = match char::from_u32(codepoint) {
+        Some(ch) => ch,
+        None => return None,
+    };
+
+    Some(Cow::Owned(ch.to_string()))
+}
+
+/// If a string starts with `&` and ends with `;`, those are removed.
+/// First trims the string.
+fn strip_entity(s: &str) -> &str {
+    let s = s.trim();
+
+    if s.starts_with('&') && s.ends_with(';') {
```

Comment:
Is there any situation where the user would expect `&copy` to do something different to `&copy;`? If not, could at least the former be permitted?

Reply:
This is true, I think we can strip these separately. I'll adjust.